model/autocomplete.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type UserAutocomplete struct {
+	In  []*User `json:"in"`
```

Comment:
So I figured out this means In or out of channel. Presumably you want to reuse this struct for in and out of team?
This was confusing to me. Maybe it would be better to call it `AutocompleteUsersInChannelResult` and the corresponding team one `AutocompleteUsersInTeamResult`. Then you can call the slices `InChannel` and `NotInChannel` or something.